### ISSUE TITLE
Conda environment installation issues

### ISSUE THREAD

**Issue by mathinic** [kuleshov-group/caduceus#44]
Hi Yair,

When attempting to install the conda environment from `caduceus_env.yml` I get the following exceptions (on multiple Ubuntu machines):
```
Pip subprocess error:
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/(...)/flash-attn_c3d53ada19464ae28f7dd98266a3a7d7/setup.py", line 9, in <module>
          from packaging.version import parse, Version
      ModuleNotFoundError: No module named 'packaging'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

failed

CondaEnvException: Pip failed
```

I'll attempt now to install the packages manually (starting with 'packaging' module), but wondered whether you see similar issues when attempting an automated conda install via `caduceus_env.yml`?

Best,
Nicolas

**Comment by yair-schiff** [comment]
I have not seen this error before unfortunately. Perhaps try updating `pip` and see if that helps (some threads online seem to indicate this could fix the issue)

**Comment by mathinic** [comment]
pip didnt do the trick, but it went through smoothly, once I manually created the conda environment, then installed first all conda dependencies and finally all pip dependencies except of the packages flash-attn, causal-conv1d and mamba-ssm. Finally I also installed those and it worked without any errors.

However I then got another issue when subsequently running `model = AutoModelForMaskedLM.from_pretrained(model_name)`:
```
RuntimeError: Failed to import transformers.generation.utils because of the following error (look up to see its traceback):
cannot import name 'split_torch_state_dict_into_shards' from 'huggingface_hub' 
```
--> once i upgraded huggingface_hub (`pip install --upgrade huggingface_hub`), this issue was also resolved. I have now huggingface-hub version 0.24.5, whereas in the original .yml file it is 0.19.4.
